### PR TITLE
Bug fixes all clients register

### PR DESCRIPTION
**Pull request by ellykits**

### CODE REVIEW

File: opensrp-chw/build.gradle [hunk 1]
```diff
@@ -178,7 +178,7 @@ android {
             buildConfigField "String[]", "ALLOWED_LOCATION_LEVELS_DEBUG", '{"Clinic" , "CHSS" , "CHA"}'
             buildConfigField "String", 'DEFAULT_LOCATION_DEBUG', '"CHA"'
             buildConfigField "String", 'DEFAULT_LOCATION', '"CHA"'
-            buildConfigField "int", "DATABASE_VERSION", '6'
+            buildConfigField "int", "DATABASE_VERSION", '8'
```
Comment (cozej4) [inline opensrp-chw/build.gradle:181]: Is there any reasons why we are skipping db version 7?

File: opensrp-chw/build.gradle [hunk 2]
```diff
@@ -163,7 +163,7 @@ android {
             buildConfigField "String[]", "ALLOWED_LOCATION_LEVELS_DEBUG", '{"Clinic" , "CHSS" , "CHA"}'
             buildConfigField "String", 'DEFAULT_LOCATION', '"CHA"'
             buildConfigField "String", 'DEFAULT_LOCATION_DEBUG', '"CHA"'
-            buildConfigField "int", "DATABASE_VERSION", '6'
+            buildConfigField "int", "DATABASE_VERSION", '8'
```
Comment (cozej4) [inline opensrp-chw/build.gradle:166]: Is there any reasons why we are skipping db version 7?

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 2
           

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/presenter/AllClientsMemberPresenter.java  6
- opensrp-chw/src/chad/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  1
- opensrp-chw/src/guinea/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  9
- opensrp-chw/src/ba/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/interactor/AllClientsMemberInteractor.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/activity/AllClientsMemberProfileActivity.java  1
- opensrp-chw/src/lmh/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  5
- opensrp-chw/src/liberia/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  12
- opensrp-chw/src/togo/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/model/AllClientsMemberModel.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/util/AllClientsUtils.java  1
         

Complexity decreasing per file
==============================
+ opensrp-chw/src/drc/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  -7
+ opensrp-chw/src/main/java/org/smartregister/chw/model/ChwAllClientsRegisterModel.java  -1
+ opensrp-chw/src/main/java/org/smartregister/chw/presenter/ChwAllClientRegisterPresenter.java  -1
         

Clones removed
==============
+ opensrp-chw/src/main/java/org/smartregister/chw/dataloader/FamilyMemberDataLoader.java  -1
+ opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyOtherMemberProfileActivity.java  -2
+ opensrp-chw/src/main/java/org/smartregister/chw/activity/AllClientsMemberProfileActivity.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5522858&bid=18094941)

File: opensrp-chw/src/main/java/org/smartregister/chw/util/AllClientsUtils.java [hunk 3]
```diff
@@ -112,4 +85,65 @@ private static Intent initProfileActivityIntent(Activity activity, CommonPersonO
...
+            intent.putExtra(Constants.INTENT_KEY.VILLAGE_TOWN, patient.getDetails().get(OpdDbConstants.KEY.HOME_ADDRESS));
+            activity.startActivity(intent);
+        }
+    }
+
+    @NotNull
+    public static List<OpdEventClient> getOpdEventClients(String jsonString) {
+        List<OpdEventClient> allClientMemberEvents = new ArrayList<>();
+
+        FamilyEventClient locationDetailsEvent = org.smartregister.family.util.JsonFormUtils.processFamilyUpdateForm(
```
Comment (keymane) [inline opensrp-chw/src/main/java/org/smartregister/chw/util/AllClientsUtils.java:116]: Codacy Issue found: [Unnecessary use of fully qualified name 'org.smartregister.family.util.JsonFormUtils.processFamilyUpdateForm' due to existing import 'org.smartregister.family.util.JsonFormUtils'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5522858)

File: opensrp-chw/src/main/java/org/smartregister/chw/model/ChwAllClientsRegisterModel.java [hunk 4]
```diff
@@ -1,125 +1,70 @@
...
-            // Inject OPenSrp id into the form
-            JSONObject stepOne = form.getJSONObject(OpdJsonFormUtils.STEP1);
-            JSONArray jsonArray = stepOne.getJSONArray(OpdJsonFormUtils.FIELDS);
-            for (int i = 0; i < jsonArray.length(); i++) {
-                JSONObject jsonObject = jsonArray.getJSONObject(i);
-                if (jsonObject.getString(OpdJsonFormUtils.KEY).equalsIgnoreCase(UNIQUE_ID)) {
-                    jsonObject.remove(OpdJsonFormUtils.VALUE);
-                    jsonObject.put(OpdJsonFormUtils.VALUE, entityId.replace("-", ""));
-                }
+    public JSONObject getFormAsJson(String formName, String entityId, String currentLocationId) {
```
Comment (keymane) [inline opensrp-chw/src/main/java/org/smartregister/chw/model/ChwAllClientsRegisterModel.java:30]: Codacy Issue found: [Avoid reassigning parameters such as 'entityId'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5522858)

File: opensrp-chw/build.gradle (hunk 1 above)
Comment (ellykits) [reply opensrp-chw/build.gradle:181]: We had  migrations for version 7 already existing in the code. It was commented out and for some reason the database version was not changed to 7. So I added the new migration for version 8 and changed the DB version accordingly.

File: opensrp-chw/build.gradle (hunk 2 above)
Comment (ellykits) [reply opensrp-chw/build.gradle:166]: Same thing here the migration code for 7 existed but was not applied to the database version on the gradle script. After adding the new migration i just bumped it to what was latest.

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/model/AllClientsMemberModel.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/util/AllClientsUtils.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/interactor/AllClientsMemberInteractor.java  1
- opensrp-chw/src/togo/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  1
- opensrp-chw/src/guinea/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  9
- opensrp-chw/src/lmh/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  5
- opensrp-chw/src/main/java/org/smartregister/chw/activity/AllClientsMemberProfileActivity.java  1
- opensrp-chw/src/ba/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  1
- opensrp-chw/src/liberia/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  12
- opensrp-chw/src/main/java/org/smartregister/chw/presenter/AllClientsMemberPresenter.java  6
- opensrp-chw/src/chad/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  1
         

Complexity decreasing per file
==============================
+ opensrp-chw/src/drc/java/org/smartregister/chw/repository/ChwRepositoryFlv.java  -7
+ opensrp-chw/src/main/java/org/smartregister/chw/presenter/ChwAllClientRegisterPresenter.java  -1
+ opensrp-chw/src/main/java/org/smartregister/chw/model/ChwAllClientsRegisterModel.java  -1
         

Clones removed
==============
+ opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyOtherMemberProfileActivity.java  -2
+ opensrp-chw/src/main/java/org/smartregister/chw/activity/AllClientsMemberProfileActivity.java  -2
+ opensrp-chw/src/main/java/org/smartregister/chw/dataloader/FamilyMemberDataLoader.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5522858&bid=18094941)

**Pull request merged**